diff -uNr knem-1.1.2/driver/linux/knem_hal.h knem-1.1.2-r1/driver/linux/knem_hal.h
--- knem-1.1.2/driver/linux/knem_hal.h	2015-02-18 20:00:09.000000000 -0200
+++ knem-1.1.2-r1/driver/linux/knem_hal.h	2015-12-07 19:15:42.575816442 -0200
@@ -301,6 +301,7 @@
 /* dma_async_memcpy_pg_to_pg removed in 3.18 together with netdma */
 #ifndef KNEM_HAVE_DMA_ASYNC_MEMCPY_PG_TO_PG
 #include <linux/dma-mapping.h>
+#include <linux/dmaengine.h>
 /* Copyright(c) 2004 - 2006 Intel Corporation. All rights reserved. */
 dma_cookie_t
 knem_dma_async_memcpy_pg_to_pg(struct dma_chan *chan, struct page *dest_pg,
