diff -uNr glusterfs-4.1.0-orig/configure.ac glusterfs-4.1.0/configure.ac
--- glusterfs-4.1.0-orig/configure.ac	2018-06-21 14:08:33.164493277 -0300
+++ glusterfs-4.1.0/configure.ac	2018-06-21 14:10:52.227393608 -0300
@@ -313,7 +313,7 @@
 dnl
 AC_ARG_WITH([ipv6-default],
         AC_HELP_STRING([--with-ipv6-default], [Set IPv6 as default.]),
-        [with_ipv6_default=${with_libtirpc}], [with_ipv6_default="no"])
+        [], [with_ipv6_default=${with_libtirpc}])
 
 if test "x$ac_cv_file__etc_redhat_release" = "xyes"; then
         if rpm -qa centos-release | grep centos; then
