diff -uNr gocryptfs-1.8.0-orig/Makefile gocryptfs-1.8.0/Makefile
--- gocryptfs-1.8.0-orig/Makefile	2020-05-09 09:20:21.000000000 -0300
+++ gocryptfs-1.8.0/Makefile	2020-05-09 23:49:58.113161770 -0300
@@ -1,7 +1,6 @@
 .phony: build
 build:
 	./build.bash
-	./Documentation/MANPAGE-render.bash
 
 .phony: test
 test:
@@ -20,6 +19,4 @@
 install:
 	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs
 	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs-xray/gocryptfs-xray
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs.1
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs-xray.1
 	install -Dm644 -t "$(DESTDIR)/usr/share/licenses/gocryptfs" LICENSE
