diff -uNr btrfs-progs-v4.9.1-r2/cmds-fi-usage.c btrfs-progs-v4.9.1-r3/cmds-fi-usage.c
--- btrfs-progs-v4.9.1-r2/cmds-fi-usage.c	2017-01-28 10:45:40.121218195 -0200
+++ btrfs-progs-v4.9.1-r3/cmds-fi-usage.c	2017-01-28 10:46:34.862545392 -0200
@@ -642,7 +642,7 @@
 	else if (ci->type & BTRFS_BLOCK_GROUP_RAID6)
 		return ci->size / (ci->num_stripes -2);
 	else if (ci->type & BTRFS_BLOCK_GROUP_RAID10)
-		return ci->size / ci->num_stripes;
+		return ci->size / (ci->num_stripes / 2);
 	return ci->size;
 }
 
