diff -uNr btrfs-progs-v5.2-orig/cmds/scrub.c btrfs-progs-v5.2/cmds/scrub.c
--- btrfs-progs-v5.2-orig/cmds/scrub.c	2019-07-05 13:21:55.000000000 -0300
+++ btrfs-progs-v5.2/cmds/scrub.c	2019-07-06 16:55:01.772430660 -0300
@@ -164,10 +164,7 @@
 		char t[4096];
 		struct tm tm;
 
-		if (s->t_resumed)
-			sec_eta = s->t_resumed;
-		else
-			sec_eta = s->t_start;
+		sec_eta = time(NULL);
 		sec_eta += sec_left;
 		localtime_r(&sec_eta, &tm);
 		t[sizeof(t) - 1] = '\0';
@@ -274,7 +271,7 @@
 		localtime_r(&ss->t_resumed, &tm);
 		strftime(t, sizeof(t), "%c", &tm);
 		t[sizeof(t) - 1] = '\0';
-		printf("Scrub resumed:   %s\n", t);
+		printf("Scrub resumed:    %s\n", t);
 	} else {
 		localtime_r(&ss->t_start, &tm);
 		strftime(t, sizeof(t), "%c", &tm);
