diff -uNr orangefs-2.9.7/Makefile.in orangefs-2.9.7-r1/Makefile.in
--- orangefs-2.9.7/Makefile.in	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/Makefile.in	2018-03-14 12:34:34.324358643 -0300
@@ -1393,7 +1393,7 @@
 src/server/simple.conf: src/apps/admin/pvfs2-genconfig
 	$(Q) "  GENCONFIG     $@"
 	$(E)$(srcdir)/src/apps/admin/pvfs2-genconfig --protocol tcp --port 3334 \
-	--ioservers localhost --metaservers localhost --logfile $(prefix)/log/pvfs2-server.log \
+	--ioservers localhost --metaservers localhost --logfile /var/log/orangefs/server.log \
 	--storage $(prefix)/storage/data --metadata $(prefix)/storage/meta \
 	--logging "server,network,storage,flow" --quiet src/server/simple.conf
 
@@ -1404,7 +1404,7 @@
 examples/fs.conf: src/apps/admin/pvfs2-genconfig
 	$(Q) "  GENCONFIG     $@"
 	$(E)$(srcdir)/src/apps/admin/pvfs2-genconfig --protocol tcp --port 3334 \
-	--ioservers localhost --metaservers localhost --logfile $(prefix)/log/pvfs2-server.log \
+	--ioservers localhost --metaservers localhost --logfile /var/log/orangefs/server.log \
 	--storage $(prefix)/storage/data --metadata $(prefix)/storage/meta \
 	--quiet examples/fs.conf
 endif
diff -uNr orangefs-2.9.7/examples/orangefs-client.in orangefs-2.9.7-r1/examples/orangefs-client.in
--- orangefs-2.9.7/examples/orangefs-client.in	2017-11-21 19:21:08.000000000 -0200
+++ orangefs-2.9.7-r1/examples/orangefs-client.in	2018-03-14 12:32:44.136657479 -0300
@@ -16,7 +16,7 @@
 OFSURL=`cat ${OFSTABFILE} | awk '{print $1}'`
 OFSMOUNT=`cat ${OFSTABFILE} | awk '{print $2}'`
 OFSOPTIONS=""
-OFSCLIENT_LOGFILE="--logfile /var/log/orangefs-client.log"
+OFSCLIENT_LOGFILE="--logfile /var/log/orangefs/client.log"
 OFSCLIENT_LOGFILE_TIMESTAMP="--logstamp=datetime"
 
 # verify presence of client binary
diff -uNr orangefs-2.9.7/src/apps/admin/pvfs2-genconfig orangefs-2.9.7-r1/src/apps/admin/pvfs2-genconfig
--- orangefs-2.9.7/src/apps/admin/pvfs2-genconfig	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/src/apps/admin/pvfs2-genconfig	2018-03-14 12:32:44.136657479 -0300
@@ -1163,7 +1163,7 @@
 
 sub get_logfile
 {
-    my $logfile = "/var/log/orangefs-server.log";
+    my $logfile = "/var/log/orangefs/server.log";
     if ($opt_logfile) {
         $logfile = $opt_logfile;
     } elsif (!$opt_quiet) {
diff -uNr orangefs-2.9.7/src/apps/kernel/linux/pvfs2-client-core.c orangefs-2.9.7-r1/src/apps/kernel/linux/pvfs2-client-core.c
--- orangefs-2.9.7/src/apps/kernel/linux/pvfs2-client-core.c	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/src/apps/kernel/linux/pvfs2-client-core.c	2018-03-14 12:32:44.137657504 -0300
@@ -88,7 +88,7 @@
 #include "pvfs2-internal.h"
 #endif
 
-#define DEFAULT_LOGFILE "/tmp/pvfs2-client.log"
+#define DEFAULT_LOGFILE "/var/log/orangefs/client.log"
 
 typedef struct
 {
diff -uNr orangefs-2.9.7/src/apps/kernel/linux/pvfs2-client.c orangefs-2.9.7-r1/src/apps/kernel/linux/pvfs2-client.c
--- orangefs-2.9.7/src/apps/kernel/linux/pvfs2-client.c	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/src/apps/kernel/linux/pvfs2-client.c	2018-03-14 12:32:44.137657504 -0300
@@ -36,7 +36,7 @@
 #define DEFAULT_ACACHE_TIMEOUT_STR "60000" /* 60 seconds */
 #define DEFAULT_NCACHE_TIMEOUT_STR "60000" /* 60 seconds */
 
-#define DEFAULT_LOGFILE "/tmp/pvfs2-client.log"
+#define DEFAULT_LOGFILE "/var/log/orangefs/client.log"
 
 #define CLIENT_RESTART_INTERVAL_SECS 10
 #define CLIENT_MAX_RESTARTS 10
@@ -270,7 +270,7 @@
                 /* catch special case of exiting due to device error */
                 if (WEXITSTATUS(ret) == (unsigned char)-PVFS_ENODEV)
                 {
-                    /* the pvfs2-client.log should log specifics from
+                    /* the client.log should log specifics from
                      * pvfs2-client-core
                      */
                     fprintf(stderr, "Device error caught, exiting now...\n");
@@ -554,7 +554,7 @@
     printf("-V, --verbose                 run in verbose output mode\n");
     printf("-f, --foreground              run in foreground mode\n");
     printf("-L  --logfile                 specify log file to write to\n"
-            "   (defaults to /tmp/pvfs2-client.log)\n");
+            "   (defaults to /var/log/orangefs/client.log)\n");
     printf("-a MS, --acache-timeout=MS    acache timeout in ms "
            "(default is %s ms)\n", DEFAULT_ACACHE_TIMEOUT_STR);
     printf("--acache-soft-limit=LIMIT     acache soft limit\n");
diff -uNr orangefs-2.9.7/src/common/misc/server-config.c orangefs-2.9.7-r1/src/common/misc/server-config.c
--- orangefs-2.9.7/src/common/misc/server-config.c	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/src/common/misc/server-config.c	2018-03-14 12:32:44.138657529 -0300
@@ -689,7 +689,7 @@
      * a particular server in the Global context.
      */
     {"LogFile",ARG_STR, get_logfile,NULL,
-        CTX_DEFAULTS|CTX_SERVER_OPTIONS,"/tmp/pvfs2-server.log"},
+        CTX_DEFAULTS|CTX_SERVER_OPTIONS,"/var/log/orangefs/server.log"},
 
     /* The LogType option can be used to control the destination of log 
      * messages from OrangeFS server.  The default value is <c>file</c>, which causes
diff -uNr orangefs-2.9.7/src/kernel/linux-2.6/super.c orangefs-2.9.7-r1/src/kernel/linux-2.6/super.c
--- orangefs-2.9.7/src/kernel/linux-2.6/super.c	2017-11-21 19:21:07.000000000 -0200
+++ orangefs-2.9.7-r1/src/kernel/linux-2.6/super.c	2018-03-14 12:32:44.138657529 -0300
@@ -1142,7 +1142,7 @@
     if (ret == -EINVAL)
     {
         gossip_err("Ensure that all pvfs2-servers have the same FS configuration files\n");
-        gossip_err("Look at pvfs2-client-core log file (typically /tmp/pvfs2-client.log) for more details\n");
+        gossip_err("Look at pvfs2-client-core log file (typically /var/log/orangefs/client.log) for more details\n");
     }
 
     if (sb)
@@ -1638,7 +1638,7 @@
         gossip_err("Ensure that all pvfs2-servers have the "
                    "same FS configuration files\n");
         gossip_err("Look at pvfs2-client-core log file "
-                   "(typically /tmp/pvfs2-client.log) for more details\n");
+                   "(typically /var/log/orangefs/client.log) for more details\n");
     }
 
     if (new_op)
